mango-viz/bdgenomics/mango/io/VCFfile.py

```diff
@@ -0,0 +1,128 @@
+
+
+    @classmethod
+    def _read(cls, filename, large = True):
+        """Open an optionally gzipped VCF file and return a pandas.DataFrame with
+        each INFO field included as a column in the dataframe.
+        
+        Borrowed from https://gist.github.com/slowkow/6215557
+
+        Note: Using large=False with large VCF files. It will be painfully slow.
+        :param filename:    An optionally gzipped VCF file.
+        :param large:       Use this with large VCF files to skip the ## lines and
+                            leave the INFO fields unseparated as a single column.
+        """
+
+        if large:
+            # Set the proper argument if the file is compressed.
+            comp = 'gzip' if filename.endswith('.gz') else None
+            # Count how many comment lines should be skipped.
+            NUM_COMMENTS = cls._count_comments(filename)
+            # Return a simple DataFrame without splitting the INFO column.
+            df = cls.dataframe_lib.read_table(filename, compression=comp, skiprows=range(NUM_COMMENTS-1))
+            NUM_SAMPLES = len(df.columns) - len(VCF_HEADER)
+            df.columns = VCF_HEADER + [i for i in range(1, NUM_SAMPLES+1)]
+            return df       
+
+    @classmethod
+    def _getcallinformation(cls, row):
+        sample_column_names = row.index[9:]
+        call_information = '"calls":['
+        for col in sample_column_names:
+            current_call = '{'
+            call_set_name = '"callSetName":"{}",'.format(str(col))
+            info = str(row[col])
+            genotype = '"genotype":["{}","{}"],'.format(info[0], info[2])
+            phaseset = '"phaseset":"True"'
+            current_call = current_call + call_set_name + genotype + phaseset + '},'
+            call_information = call_information + current_call
+        call_information = call_information[:-1] + ']'
+        return call_information
+    
+    
+    @classmethod
+    def _parse(cls, df):
+        def _buildrow(row):
+            reference_name = row["CHROM"]
+            start = row["POS"]
+            end = start + 1
+            reference_bases = row["REF"]
+            alternate_bases = row["ALT"]
+            ids = row["ID"]
+            call_information = cls._getcallinformation(row)
+            content = '"referenceName":"{}", "start":"{}", "end":"{}", "referenceBases":"{}", "alternateBases":"{}", "id":"{}", {}'.format(str(reference_name), str(start), str(end), str(reference_bases), str(alternate_bases), str(ids), call_information)
+            return content
+        return df.apply(_buildrow, axis = 1).squeeze()
+
+    @classmethod
+    def _to_json(cls, df):
+        content_series = cls._parse(df)
+        json_ga4gh = "{\"variants\":["
+        json_ga4gh += "{" + content_series.str.cat(sep="},{")[:-2]
+        json_ga4gh =  json_ga4gh + "}]}]}"
+        
+        return json_ga4gh
+        
+    
+    @classmethod
+    def _visualization(cls, df):
+        return 'variantJson'
```

Comment:
This is already defined by GA4GHVariantJson.name in sources.py, so you can pull the string from there